org.adempiere.base/src/org/compiere/process/ProcessInfoParameter.java

```diff
@@ -282,4 +300,134 @@ public void setParameterName (String ParameterName)
+	 * Return the value of the parameter as a validated String (all values between commas must be integer)
+	 * @return String
+	 */
+	private String getParameterAsCSVInt(String param) {
+
+		if (Util.isEmpty(param))
+			return "";
+
+		String[] strarr = param.split(",");
+
+		for (String par : strarr)
+			Integer.valueOf(par);
+
+		return param;
+	}
+
+	/**
+	 * Return the value of the parameter as a String with all values between commas surrounded by quotes
+	 * @return String
+	 */
+	public String getParameterAsCSVString() {
+		return getParameterAsCSVString(getParameterAsString());
+	}
+
+	/**
+	 * Return the value of the parameter as a String with all values between commas surrounded by quotes
+	 * @return String
+	 */
+	public String getParameter_ToAsCSVString() {
+		return getParameterAsCSVString(getParameter_ToAsString());
+	}
+
+	/**
+	 * Return the value of the parameter as a String with all values between commas surrounded by quotes
+	 * @return String
+	 */
+	private String getParameterAsCSVString(String param) {
+		if (Util.isEmpty(param))
+			return "";
+
+		String[] strarr = ((String) param).split(",");
+
+		StringBuilder whereValidated = new StringBuilder();
+		for (String par : strarr) {
+			if (whereValidated.length() > 0)
+				whereValidated.append(",");
+			whereValidated.append(DB.TO_STRING(par));
+		}
+
+		return whereValidated.toString();
+	}
+
+	/**
+	 * Return the value of the parameter as an array of int. Validate every value is an integer and throws NumberFormatException if one of the value is not a valid integer.
+	 * @return String
+	 */
+	public int[] getParameterAsIntArray() {
+		return getParameterAsIntArray(getParameterAsString());
+	}
+
+	/**
+	 * Return the value of the parameter To as an array of int. Validate every value is an integer and throws NumberFormatException if one of the value is not a valid integer.
+	 * @return String
+	 */
+	public int[] getParameterToAsIntArray() {
+		return getParameterAsIntArray(getParameter_ToAsString());
+	}
+
+	/**
+	 * Return the value of the parameter as an array of int. Validate every value is an integer and throws NumberFormatException if one of the value is not a valid integer.
+	 * @return String
+	 */
+	private int[] getParameterAsIntArray(String param) {
+
+		if (Util.isEmpty(param))
+			return new int[0];
+
+		return Arrays.stream(param.split(",")).mapToInt(Integer::parseInt).toArray();  
+	}
+
+	/**
+	 * Return the value of the parameter as an array of String.
+	 * @return String
+	 */
+	public String[] getParameterAsStringArray() {
+		return getParameterAsStringArray(getParameterAsString());
+	}
+
+	/**
+	 * Return the value of the parameter To as an array of String.
+	 * @return String
+	 */
+	public String[] getParameterToAsStringArray() {
+		return getParameterAsStringArray(getParameter_ToAsString());
+	}
+
+	/**
+	 * Return the value of the parameter as an array of String.
+	 * @return String
+	 */
+	private String[] getParameterAsStringArray(String param) {
+

```

Comment:
sorry to be picky here but there are still several method above with wrong "@return" type

Reply:
should be ok now :)